src/mpi/coll/allreduce.c

```diff
@@ -649,8 +649,12 @@ int MPIR_Allreduce_reduce_scatter_allgather(
 
     /* find nearest power-of-two less than or equal to comm_size */
     pof2 = 1;
-    while (pof2 <= comm_size) pof2 <<= 1;
-    pof2 >>=1;
+    if (comm_size == 1)
+        iter = 0;
+    else
+        iter = -1;
+    while (pof2 <= comm_size) { iter ++; pof2 <<= 1;}
+    pof2 >>= 1;
```

Comment:
This would set iter=1 when comm_size=1, is that intentional? iter should be zero when comm_size=1.